File: python/LongestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        str = ''
        if len(strs) == 0:
            return str
        min = len(strs[0])
        for i in range(len(strs)):
            if len((strs[i]))<min:
                min = len((strs[i]))
        
        for j in range(min):
            a = []
            for i in range(len((strs))):
                a.append(strs[i][j])
            if len(set(a)) == 1:
                str +=a[0]
            else:
                break
        return str

File: cpp/LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string res = "";
        if (strs.empty()) 
            return res;        
        for (int j = 0; j < strs[0].size(); ++j) {
            char c = strs[0][j];
            for (int i = 1; i < strs.size(); ++i) {
                if (j >= strs[i].size() || strs[i][j] != c) {
                    return res;
                }
            }
            res.push_back(c);
        }
        return res;
    }
};
